src/octoprint/server/api/folders.py

```diff
@@ -0,0 +1,85 @@
+# coding=utf-8
```

Comment:
Hm... I'd rather not have a completely new sub API just for this to be honest, especially not with an arbitrary name like `/folders` that could just as well refer to uploaded files. IMHO fetching information about the system such as this could be located either underneath the `/settings` endpoint or maybe the `/system` endpoint (e.g. `/system/folders`).

I'd also like to see a "fetch all folder usage stats" endpoint, otherwise the frontend will have to do quite a number of requests just to be able to display consistent information. We can still have sub resources for the individual folders, but I'd first and foremost concentrate on a catch all.

Reply:
That makes sense. `/system/folders` sounds like a good place to me.